### DISCUSSION TITLE
Add texture to dielectric material

### DISCUSSION THREAD

**Discussion by duongnlt** [mitsuba-renderer/mitsuba3#1112] [Q&A]
Hi team,

I have a setup with a dielectric object and a polarizer in front of a camera. I'm interested in adding a texture file, similar to the one provided, to the dielectric material. Is there a method to accomplish this?
![marble_44_basecolor-1K](https://github.com/mitsuba-renderer/mitsuba3/assets/44237568/9ccdbd7e-c49b-4de7-b37b-14e787f43ad8)

**Comment by njroussel** [comment]
Hi @duongnlt 

I don't believe we have ship a suitable BSDF for such a material.

You can have a textured transmissive material with the `principled` BSDF, but it does not support polarization.
The only option would be to build your own custom BSDF by adding a tint, and maybe a spatially varying roughness parameter. This will be fairly complex as you need the polarization state to be handled correctly too.
